packages/web-config-server/src/apiV1/dataBuilders/generic/table/tableOfDataValues/helpers/getValuesByCell.js

```diff
@@ -49,7 +51,9 @@ export const getValuesByCell = (config, results) => {
   return zipObject(
     Object.keys(groupedResults),
     Object.values(groupedResults).map(([{ value, metadata }]) =>
-      metadata && metadata.options ? metadata.options[value] : value,
+      metadata && metadata.options && !isEqual(metadata.options, DEFAULT_BINARY_OPTIONS_OBJECT) // If the metadata options is the default Yes/No, return default value 1/0 instead of translating value.
```

Comment:
Nice one!

Could you add more context on this comment. Just so that someone else can also understand the underlying reason why we are not translating default binary options here.

```suggestion
      metadata && metadata.options && !isEqual(metadata.options, DEFAULT_BINARY_OPTIONS_OBJECT) // metadata options are mainly used for translating data from DHIS2 that has category options. For internal data, it returns default Yes/No metadata options for binary data to support translating from 1/0 to Yes/No in a few other places. For matrices, particularly `tableOfDataValues` data builder, we want to use raw 1/0 values and don't want to apply this translation because we want to keep the values consistent. So this check is for disabling that special case.
```